client/engine/chainservice/adjudicator/challenge_fevm_test.go

```diff
@@ -0,0 +1,165 @@
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+	"math/rand"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	hdwallet "github.com/miguelmota/go-ethereum-hdwallet"
+	"github.com/statechannels/go-nitro/channel/state"
+	"github.com/statechannels/go-nitro/channel/state/outcome"
+
+	"github.com/statechannels/go-nitro/types"
+)
+
+// This is the mnemonic for the prefunded accounts on wallaby.
+// The first 25 accounts will be prefunded.
+const WALLABY_MNEMONIC = "army forest resource shop tray cluster teach cause spice judge link oppose"
+
+// This is the HD path to use when deriving accounts from the mnemonic
+const WALLABY_HD_PATH = "m/44'/1'/0'/0"
+
+func TestChallengeFEVM(t *testing.T) {
+	t.Skip() // We only want to run this test manually for now.
+	hyperspaceChainId := big.NewInt(3141)
+
+	// Setup transacting EOA
+	wallet, err := hdwallet.NewFromMnemonic(WALLABY_MNEMONIC)
+	if err != nil {
+		panic(err)
+	}
+
+	// The 0th account is usually used for deployment so we grab the 1st account
+	a, err := wallet.Derive(hdwallet.MustParseDerivationPath(fmt.Sprintf("%s/%d", WALLABY_HD_PATH, 1)), false)
+	if err != nil {
+		panic(err)
+	}
+
+	//PK: 0x1688820ffc6a811e09ff17eccec23d8dec4850c3098ffc03ac4aa38dd8f3a994
+	// corresponding ETH address is 0x280c53E2C574418D8d6d8d651d4c3323F4b194Be
+	// corresponding f4 address (delegated) is t410ffagfhywforay3dlnrvsr2tbtep2ldff6xuxkrjq.
+	pk, err := wallet.PrivateKey(a)
+
+	if err != nil {
+		panic(err)
+	}
+	client, err := ethclient.Dial("https://api.hyperspace.node.glif.io/rpc/v0")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// When submitting a transaction it's signed against a specific chain id
+	// To get the correct signature we need to use the correct chain id that wallaby is expecting
+	txSubmitter, err := bind.NewKeyedTransactorWithChainID(pk, hyperspaceChainId)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// By setting the GasTipCap we signal this is a type 2 transaction
+	// FEVM does NOT support type 1 transactions
+	txSubmitter.GasTipCap = big.NewInt(300000)
+
+	// This is the deployed contract on wallaby
+	// If wallaby gets reset this will need to be redeployed by running:
+	// WALLABY_DEPLOYER_PK="f4d69c36885541f56f4728ddc002a6fa2fcb26c9f608910310a776c83b7fde47" npx hardhat deploy --network hyperspace --deploy-scripts ./hardhat-deploy-fvm --reset
+	// The PK corresponds to account 0xE39dce95b1A924E2472E24C20C55eA3559a09251.
+	// It should be prefunded after every wallaby reset.
+	naAddress := common.HexToAddress("0x4fBeCDA4735eaF21C8ba5BD40Ab97dFa2Ed88E80")
+	caAddress := common.HexToAddress("0xC57875E317f67F2bE5D62f5c7C696D2eb7Fe79FE")
+
+	na, err := NewNitroAdjudicator(naAddress, client)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	balance := new(big.Int)
+	balance.SetString("10000000000000000000", 10) // 10 eth in wei
+
+	TestChallengeWithTurnNum := func(t *testing.T, turnNum uint64) {
+
```

Comment:
Nit, seems like our code base mixes zero or one new line after a definition. We should probably stick with one standard.

Reply:
Agreed. Do we have a preference?

Reply:
I prefer no newlines.